### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by greypants**
I fell in love with Webpack!

– Super smart automatic code splitting
– Async requires for free! 
– More robust in general than Browserify

Live deployed result (open inspector, and look at the network tab and the console): 
http://greypants.github.io/gulp-starter/

Async Screencapture: 
http://f.cl.ly/items/352C3m2w2c342k040A0T/async.mp4

### CODE REVIEW

**Comment by greypants** [comment]
#TODO: fix travis ci integration

**Comment by solomonhawk** [comment]
Did you choose to break up the gulp configuration because the one file was becoming too large? My initial thought is that I would rather have 1 file to look at when modifying gulp config.

:+1:

**Comment by greypants** [comment]
@solomonhawk yeah, it was getting pretty huge with the addition of webpack, and there's logic and dependencies in a lot of them - more than just lists of settings and paths.

File: gulpfile.js/tasks/webpack-development.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+var gulp         = require('gulp')
+var logger       = require('../lib/compileLogger')
+var webpack      = require('webpack')
+var browserSync  = require('browser-sync')
+
+gulp.task('webpack:development', function(callback) {
+  var built = false
+
+  webpack(config).watch(200, function(err, stats) {
+    logger(err, stats)
```
Comment (averyvery) [inline gulpfile.js/tasks/webpack-development.js]: Totally need to steal this.

File: gulpfile.js/config/webpack.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+var paths           = require('./')
+var webpack         = require('webpack')
+var webpackManifest = require('../lib/webpackManifest')
+
+module.exports = function(env) {
```
Comment (averyvery) [inline gulpfile.js/config/webpack.js]: This is great!

**Comment by averyvery** [comment]
:+1:

File: gulpfile.js/tasks/build-production.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+var gulp         = require('gulp');
+var gulpSequence = require('gulp-sequence');
+
+gulp.task('build:production', function(cb) {
+  process.env.NODE_ENV = 'production'
```
Comment (greypants) [inline gulpfile.js/tasks/build-production.js]: Oops. Don't need this line anymore.

File: gulpfile.js/tasks/rev/rev-css.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+var config = require('../../config');
+var filter = require('gulp-filter');
+var gulp   = require('gulp');
+var minify = require('gulp-minify-css');
+var rev    = require('gulp-rev');
+var uglify = require('gulp-uglify');
+
+// 4) Rev and compress CSS and JS files (this is done after assets, so that if a
```
Comment (greypants) [inline gulpfile.js/tasks/rev/rev-css.js]: Not doing JS files here anymore

**Comment by nhunzaker** [comment]
:+1:

**Comment by tommymarshall** [comment]
This is $$$$$ :+1: 
One style thing, sometimes you use `"`'s and sometimes `'`'s others.

**Comment by greypants** [comment]
@tommymarshall will do a global find and replace, and also remove semicolons :)

**Comment by nhunzaker** [comment]
@greypants :heart_eyes:

**Comment by ieugen** [comment]
> I fell in love with Webpack! 
@greypants I won't mention broccoli  . You might have to change the name of the project :)

**Comment by greypants** [comment]
@ieugen lol.

**Pull request merged**